src/src/PluginStructs/P128_data_struct.cpp

```diff
@@ -1432,39 +1438,67 @@ void P128_data_struct::NeoPixelSendStatus(struct EventStruct *eventSource) {
-  json += Plugin_128_pixels->GetBrightness();
-  json += F("\",\n\"rgb\": \"");
-  json += colorStr;
-  json += F("\",\n\"hue\": \"");
-  json += toString((HsbColor(RgbColor(rgb.R, rgb.G, rgb.B)).H * 360), 0);
-  json += F("\",\n\"saturation\": \"");
-  json += toString((HsbColor(RgbColor(rgb.R, rgb.G, rgb.B)).S * 100), 0);
-  json += F("\",\n\"brightness\": \"");
-  json += toString((HsbColor(RgbColor(rgb.R, rgb.G, rgb.B)).B * 100), 0);
-  json += F("\",\n\"bgcolor\": \"");
-  json += backgroundcolorStr;
-  json += F("\",\n\"count\": \"");
-  json += count;
-  json += F("\",\n\"speed\": \"");
-  json += speed;
-  json += F("\",\n\"pixelcount\": \"");
-  json += pixelCount;
-  json += F("\"\n}\n");
-  SendStatus(eventSource, json); // send http response to controller (JSON format)
+  json += F("{\n");                                                       // 2
+  json += F("\"plugin\": \"128");                                         // 14
+  json += F("\",\n\"mode\": \"");                                         // 12
+  json += P128_modeType_toString(mode);                                   // 2-11
+  json += F("\",\n\"lastmode\": \"");                                     // 16
+  json += P128_modeType_toString(savemode);                               // 2-11
+  json += F("\",\n\"fadetime\": \"");                                     // 16
+  json += fadetime;                                                       // 1-5
+  json += F("\",\n\"fadedelay\": \"");                                    // 16
+  json += fadedelay;                                                      // 1-5
+  json += F("\",\n\"dim\": \"");                                          // 9
+  json += Plugin_128_pixels->GetBrightness();                             // 1-3
+  json += F("\",\n\"rgb\": \"");                                          // 9
+  json += colorStr;                                                       // 6-8
+  json += F("\",\n\"hue\": \"");                                          // 9
+  json += toString((HsbColor(RgbColor(rgb.R, rgb.G, rgb.B)).H * 360), 0); // 8?
+  json += F("\",\n\"saturation\": \"");                                   // 18
+  json += toString((HsbColor(RgbColor(rgb.R, rgb.G, rgb.B)).S * 100), 0); // 8?
+  json += F("\",\n\"brightness\": \"");                                   // 18
+  json += toString((HsbColor(RgbColor(rgb.R, rgb.G, rgb.B)).B * 100), 0); // 8?
+  json += F("\",\n\"bgcolor\": \"");                                      // 15
+  json += backgroundcolorStr;                                             // 6-8
+  json += F("\",\n\"count\": \"");                                        // 13
+  json += count;                                                          // 1-3
+  json += F("\",\n\"speed\": \"");                                        // 13
+  json += speed;                                                          // 1-2
+  json += F("\",\n\"pixelcount\": \"");                                   // 18
```

Comment:
We do have functions to wrap some string in characters and also for json like elements.

Reply:
I need to wrap my head around this a bit more before I grasp this, and try to improve.

Reply:
Used `to_json_object_value` function to improve the code, this also implies that numeric values are no longer quoted, as that is against JSON standards. If needed we could apply the Advanced setting for quoted `bool` values here also, but unless someone is using a home-crafted, incorrect, JSON parser, this shouldn't be a problem.